test/cloud-slack-dev-e2e/botkube_page_helpers_test.go

```diff
@@ -0,0 +1,215 @@
+	t.Log("Log into Botkube Cloud Dashboard")
+
+	p.page.Page = page
+
+	p.page.MustNavigate(appendOrgIDQueryParam(t, p.cfg.BotkubeCloud.UIBaseURL, p.cfg.BotkubeCloud.TeamOrganizationID))
+	p.page.MustWaitNavigation()
+
+	p.page.MustElement(`input[name="username"]`).MustInput(p.cfg.BotkubeCloud.Email)
+	p.page.MustElement(`input[name="password"]`).MustInput(p.cfg.BotkubeCloud.Password)
+	p.page.MustElementR("button", "^Continue$").MustClick()
+	p.page.Screenshot()
+}
+
+func (p *BotkubeCloudPage) HideCookieBanner(t *testing.T) {
+	t.Log("Hide Botkube cookie banner")
+	p.page.MustElementR("button", "^Decline$").MustClick()
+	p.page.Screenshot()
+}
+
+func (p *BotkubeCloudPage) CaptureBearerToken(t *testing.T, browser *rod.Browser) func() {
+	t.Logf("Starting hijacking requests to %q to get the bearer token...", p.GQLEndpoint)
+
+	router := browser.HijackRequests()
+	router.MustAdd(p.GQLEndpoint, func(ctx *rod.Hijack) {
+		if p.AuthHeaderValue != "" {
+			ctx.ContinueRequest(&proto.FetchContinueRequest{})
+			return
+		}
+
+		if ctx.Request != nil && ctx.Request.Method() != http.MethodPost {
+			ctx.ContinueRequest(&proto.FetchContinueRequest{})
+			return
+		}
+
+		require.NotNil(t, ctx.Request)
+		p.AuthHeaderValue = ctx.Request.Header(authHeaderName)
+		ctx.ContinueRequest(&proto.FetchContinueRequest{})
+	})
+	go router.Run()
+	return router.MustStop
+}
+
+func (p *BotkubeCloudPage) CreateNewInstance(t *testing.T, name string) {
+	t.Log("Create new Botkube Instance")
+
+	p.page.MustElement("h6#create-instance").MustClick()
+	p.page.MustElement(`input[name="name"]`).MustSelectAllText().MustInput(name)
+	p.page.Screenshot()
+
+	// persist connected deploy info
+	_, id, _ := strings.Cut(p.page.MustInfo().URL, "add/")
+	p.ConnectedDeploy = &gqlModel.Deployment{
+		Name: name,
+		ID:   id,
+	}
+}
+
+func (p *BotkubeCloudPage) InstallAgentInCluster(t *testing.T, botkubeBinary string) {
+	t.Log("Installing Botkube using Botkube CLI")
+
+	installCmd := p.page.MustElement("div#install-upgrade-cmd > kbd").MustText()
+
+	args, err := shellwords.Parse(installCmd)
+	args = append(args, "--auto-approve")
+	require.NoError(t, err)
+
+	cmd := exec.Command(botkubeBinary, args[1:]...)
+	installOutput, err := cmd.CombinedOutput()
+	t.Log(string(installOutput))
+	require.NoError(t, err)
+
+	p.page.MustElement("button#cluster-connected").MustClick()
+}
+
+func (p *BotkubeCloudPage) OpenSlackAppIntegrationPage(t *testing.T) {
+	p.page.MustElement(`button[aria-label="Add tab"]`).MustClick()
+	p.page.MustWaitStable()
+	p.page.MustElementR("button", "^Slack$").MustClick()
+	p.page.MustWaitStable()
+	p.page.Screenshot()
+
+	p.page.MustElementR("a", "Add to Slack").MustClick()
+}
+
+// ReAddSlackPlatformIfShould add the slack platform again as the page was often not refreshed with a newly connected Slack Workspace.
+// It only occurs with headless mode.
+// TODO(@pkosiec): Do you have a better idea how to fix it?
```

Comment:
@pkosiec

Reply:
Nope, unfortunately I don't have any idea here 😞 in theory tab focus should trigger refetching data but looks like in headless mode it is not detected properly 🤔